source/common/router/config_impl.cc

```diff
@@ -783,7 +783,8 @@ void RouteEntryImplBase::finalizeRequestHeaders(Http::RequestHeaderMap& headers,
   for (const HeaderParser* header_parser : getRequestHeaderParsers(
            /*specificity_ascend=*/vhost_.globalRouteConfig().mostSpecificHeaderMutationsWins())) {
     // Later evaluated header parser wins.
-    header_parser->evaluateHeaders(headers, stream_info);
+    header_parser->evaluateHeaders(headers, headers,
+                                   *Http::StaticEmptyHeaders::get().response_headers, stream_info);
```

Comment:
To https://github.com/envoyproxy/envoy/pull/21932#discussion_r970243905

For example, change of here can be eliminated by the wrapper, right?

Reply:
Added a wrapper with 2 params and request headers are taken from stream_info.